test/e2e/conditions_test.go

```diff
@@ -3,11 +3,36 @@
 package e2e
 
 import (
+	"fmt"
+
+	templatev1 "github.com/openshift/api/template/v1"
 	authorizationapiv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func (t *testClient) templateInstanceIsReady() (bool, error) {
```

Comment:
should probably take the namespace and name as arguments if it's going in a different file

Reply:
That would not work because the fn signature here needs to be a `wait.ConditionFunc` which is a `func() (bool, error)` in order to be used as an input in `wait.PollImmediate`. I guess you could pass a closure in PollImmediate but I don't feel strong about generalizing yet. `conditions_test.go` holds all the ConditionFuncs we use in our e2es but they are not necessarily reusable across tests and I am not sure most of them are going to be used more than once, it's just a clean way to separate tests from helpers.

Reply:
I could also put the template instance reference in testClient before polling, added benefit would be I don't have to query for the template instance

Reply:
Let's keep the testClient hold just clients and not resource references. Also, you will have to query for the template instance anyway inside the condition func in order to determine whether its status is ready.